homeassistant/exceptions.py

```diff
@@ -21,11 +23,30 @@ def __init__(
         translation_placeholders: dict[str, str] | None = None,
     ) -> None:
         """Initialize exception."""
+        if not args and translation_key and translation_domain:
+            self._translate = True
+            args = (translation_key,)
+
         super().__init__(*args)
         self.translation_domain = translation_domain
         self.translation_key = translation_key
         self.translation_placeholders = translation_placeholders
 
+    def __str__(self) -> str:
+        """Return exception message string from translation cache."""
+        if not self._translate:
+            return super().__str__()
```

Comment:
Maybe like this?
```suggestion
        if self.args:
            return super().__str__()
        if self._message:
            return self._message
```

Reply:
I'd like us to keep `_translate` so that it can be set at class level for sub classes.